toolchain/docs/adding_features.md

```diff
@@ -261,13 +268,38 @@ If the resulting SemIR needs a new instruction:
     };
     ```
 
-Adding an instruction will also require a handler in the Lower step.
+    -   [`sem_ir/inst_kind.h`](/toolchain/sem_ir/inst_kind.h) documents the
+        different options when defining a new instruction, as well as their
+        defaults, see `InstKind::DefinitionInfo`.
+    -   If an instruction doesn't have a `type_id`, it won't produce a value,
+        and so won't be given a label when formatted, and can't be the argument
+        of another instruction.
+    -   If an instruction always produces a type, so it sets
+        `.is_type = InstIsType::Always`, then its `type_id` field will be set to
+        `SemIR::TypeId::TypeType`.
```

Comment:
`.is_type` indicates whether the `type_id` is set to `TypeType` (`Never`/`Maybe`/`Always`), but it doesn't cause the value of `type_id` to be set; instead, it mainly adds safeguards and changes some lowering logic.

What's the nuance you're trying to get at here? Above there's the reference to inst_kind.h, which has more detail.

Reply:
I've revised the text to hopefully make my point more clearly.